R/sim_gs_n.R

```diff
@@ -242,15 +233,29 @@ sim_gs_n <- function(
     block = rep(c("experimental", "control"), 2),
     test = wlr,
     cut = NULL,
-    seed = 2024,
     ...) {
   # Input checking
   # TODO
 
+  # parallel computation message for backends ----
+  if (!is(plan(), "sequential")) {
+    # future backend
+    message("Using ", nbrOfWorkers(), " cores with backend ", attr(plan("list")[[1]], "class")[2])
+  } else if (foreach::getDoParWorkers() > 1) {
+    message("Using ", foreach::getDoParWorkers(), " cores with backend ", foreach::getDoParName())
+    message("Warning: ")
+    message("doFuture may exhibit suboptimal performance when using a doParallel backend.")
+  } else {
+    message("Backend uses sequential processing.")
+  }
```

Comment:
We should add test cases for both the doFuture and doParallel backends. I'm pretty sure the GitHub Actions machines have 2 cores available. And if we're worried about CRAN, we can always add `skip_on_cran()` for these tests of the parallelization method

Reply:
I agree on adding tests. CRAN enforces the max 2 core policy strictly so to avoid any potential trouble, always using `skip_on_cran()` is a good idea, just in case someone _accidentally_ added something that used all cores.